### PR TITLE
Change to combine restoration prologue with coverage register spill

### PR DESCRIPTION
**Pull request by WorksButNotTested**
Please can you give this branch a try and see if it improves performance any for you? Thanks.

### CODE REVIEW

**Review by s1341 — APPROVED** [review]
LGTM

**Comment by tokatoka** [comment]
This has broke windows example :<

**Comment by tokatoka** [comment]
I get STATUS_ILLEGAL_INSTRUCTION now with harness_win.cpp example
You can reproduce with `cargo make test` in frida_libpng

previously it is STATUS_ACCESS_VIOLATION (This is intended)

File: libafl_frida/src/coverage_rt.rs [hunk 1]
```diff
@@ -184,6 +187,24 @@ impl CoverageRuntime {
...
+        // be used, then this instruction is simply skipped and Stalker simply
+        // branches to the second instruction in the block.
+        //
+        // Since we also need to spill some registers in order to update our
+        // coverage map, in the event of a long branch, we can simply re-use
+        // these spilt registers. This, however, means we need to retard the
+        // code writer so that we can overwrite the so-called "restoration
+        // prologue".
+        let pc = writer.pc();
+        writer.reset(pc - 4);
```
Comment (tokatoka) [inline libafl_frida/src/coverage_rt.rs:206]: Perhaps instruction size is not (always) 4 on amd64?

Comment (WorksButNotTested) [reply libafl_frida/src/coverage_rt.rs:206]: My bad. That was obviously only supposed to happen on AARCH64. Let me put together a patch. I can test on Linux x64, but I don't have windows. Are you able to check for me?

Comment (tokatoka) [reply libafl_frida/src/coverage_rt.rs:206]: Yes of course 😉 
... Well in fact the ci will turn green if it works

**Comment by domenukk** [comment]
Maybe we should start adding a step that looks for existing corpus entries/crashes to the test case makefiles

**Comment by tokatoka** [comment]
It's for windows only, but we are doing it already for frida_libpng.

and thanks to it, the ci failed and I noticed the error

**Pull request merged**